core/trino-main/src/main/java/io/trino/operator/join/PagesHash.java

```diff
@@ -30,6 +30,24 @@
 
     int getAddressIndex(int rightPosition, Page hashChannelsPage, long rawHash);
 
+    default int[] getAddressIndex(int[] positions, Page hashChannelsPage)
+    {
+        int[] result = new int[positions.length];
```

Comment:
nit: consider capping batch size for better CPU cache utilization

Reply:
No difference in benchmarks